airsonic-main/src/main/java/org/airsonic/player/spring/RegisterPrecompiledJSPInitializer.java

```diff
@@ -0,0 +1,65 @@
+package org.airsonic.player.spring;
+
+import org.airsonic.player.service.SettingsService;
+import org.apache.commons.io.IOUtils;
+import org.apache.tomcat.util.descriptor.web.ServletDef;
+import org.apache.tomcat.util.descriptor.web.WebXml;
+import org.apache.tomcat.util.descriptor.web.WebXmlParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.web.servlet.ServletContextInitializer;
+import org.springframework.stereotype.Component;
+import org.xml.sax.InputSource;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletRegistration;
+
+import java.io.InputStream;
+import java.io.SequenceInputStream;
+import java.nio.charset.Charset;
+import java.util.Map;
+
+@Component
+public class RegisterPrecompiledJSPInitializer implements ServletContextInitializer {
+
+    private static final Logger logger = LoggerFactory.getLogger(RegisterPrecompiledJSPInitializer.class);
+
+    @Override
+    public void onStartup(ServletContext servletContext) {
+
+        if(!SettingsService.isDevelopmentMode()) {
+            logger.debug("Registering precompiled jsps");
+            registerPrecompiledJSPs(servletContext);
+        } else {
+            logger.debug("Not registering precompiled jsps");
+        }
+    }
+
+    private static void registerPrecompiledJSPs(ServletContext servletContext) {
+        WebXmlParser parser = new WebXmlParser(false, true, true);
+        InputStream precompiledJspWebXml = RegisterPrecompiledJSPInitializer.class.getResourceAsStream("/precompiled-jsp-web.xml");
+        InputStream webXmlIS = new SequenceInputStream(
+                new SequenceInputStream(
+                        IOUtils.toInputStream("<!DOCTYPE web-app PUBLIC " +
+                                "\"-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN\" " +
+                                "\"http://java.sun.com/dtd/web-app_2_3.dtd\"><web-app>", Charset.defaultCharset()),
+                        precompiledJspWebXml),
+                IOUtils.toInputStream("</web-app>", Charset.defaultCharset()));
+        WebXml webXml = new WebXml();
+        boolean success = parser.parseWebXml(new InputSource(webXmlIS), webXml, false);
+        if (!success) {
+            throw new RuntimeException("Error parsing precompiled jsp XML " + precompiledJspWebXml);
+        }
+        for (ServletDef def :  webXml.getServlets().values()) {
+            logger.trace("Registering precompiled JSP: {} -> {}", def.getServletName(), def.getServletClass());
+            ServletRegistration.Dynamic reg = servletContext.addServlet(def.getServletName(), def.getServletClass());
+            reg.setLoadOnStartup(99);
```

Comment:
Why this magic constant?

Reply:
See https://stackoverflow.com/questions/809775/what-does-the-servlet-load-on-startup-value-signify. Basically it needs to be between 0 and 128. The 99 I believe is so that these servlets dont take precedence over more important things.

Reply:
Can you add a small comment? :)